### DISCUSSION TITLE
dead loop cannot be traced

### DISCUSSION THREAD

**Discussion by jiandongg** [cocotb/cocotb#3891] [Q&A] [answered]
when I write a dead loop in my cocotb test like that:
```python
while 1:
        await RisingEdge(dut.m_axi_aclk)
```
the python code will stuck before the dead loop part, like:
```python
    print("AAA")
    rx_frame = await tb.cmac_sink.recv()
    print("BBB")
```

In the terminal, I can see py print "AAA", not print "BBB", it looks like cocotb stuck in `rx_frame = await tb.cmac_sink.recv()`. but actually, py is stuck in 
```python
while 1:
        await RisingEdge(dut.m_axi_aclk)
```
after the `rx_frame = await tb.cmac_sink.recv()`

That makes me confused. Can you help me?

**Comment by ktbarrett** [comment, marked as answer]
Prints are not flushed by default, which is why you aren't seeing it print `BBB`. Use `print(...., flush=True)`.

**Reply by jiandongg** [reply]
Thank you so much!

Use "flush=True" really worked. I can trace the stuck point more easily :-D

Really appreciated your help, and projects like cocotb, python3, etc🌹
